### ISSUE TITLE
[java] UselessStringValueOf: False positive when there is no initial String to append to

### ISSUE THREAD

**Issue by pzygielo** [pmd/pmd#3492]
**Affects PMD Version:** 6.37.0

**Rule:** [UselessStringValueOf](https://pmd.github.io/latest/pmd_rules_java_performance.html#uselessstringvalueof)

**Description:** False positive, when there is no initial String to append to.

**Code Sample demonstrating the issue:**
40b5f95f3adb9d94b3bf0119803f9d8c39b88c5e/https://github.com/pmd/pmd/pull/3490
```java
public class Test {
    private short getPrefix(long id) {
        return 0;
    }
    private short getTimestamp(long id) {
        return 0;
    }
    private short getCounter(long id) {
        return 0;
    }
    public String toShortString(long id) {
        return String.valueOf(getPrefix(id)) + getTimestamp(id) + getCounter(id); // (12)
    }
}
```

**Expected outcome:**

PMD reports a violation at line 12, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*
